core/src/core_tests/local_activities.rs

```diff
@@ -973,7 +973,8 @@ async fn resolved_las_not_recorded_if_wft_fails_many_times() {
                 ..Default::default()
             })
             .await;
-            panic!("Oh nooooo")
+            let _ = "panic".parse::<i32>().unwrap();
+            Ok(().into())
```

Comment:
This test needs to stay as a panic, it's important to the goal of the test. You'll have to just add a type hint for the closure's return type.

Reply:
I tried a lot to replace 'Payload' the Workflow Future with a trait before starting the discussion 😄.
It ends with the requirement of WorkflowFunction becoming generic and then bounds won't add up correctly and the From implementation fails with 'unconstrained' bounds etc. I will try again though.

Regarding the Payload being generic. I changed the trait a bit for now [here](https://github.com/temporalio/sdk-core/commit/bb612333dc4cdc11e4db4999e2109fb9a00f8659), next PR. I think we need to pass the Payload converter into both WfContext and ActContext and use it in a wrapper function like [this](https://github.com/h7kanna/sdk-core/blob/payload-converter/sdk/src/lib.rs#L765).

I will work on how we can make this dynamic without help from the generics. Currently looks like this

```move |ctx: WfContext| async move {
    let activity = ActivityOptions {
        activity_id: Some(activity_id.to_string()),
        ..Default::default()
    };
    let result = ctx.activity(activity).await.unwrap_ok_payload();
    let result = String::from_payload(None, &result).expect("serializes fine");
    Ok(WfExitValue::Normal(result))
}
```

Reply:
Type inference won't work and the code becomes unreachable after `panic!()`. So panicking with unwrap

Reply:
Ah right, I missed the unwrap - regardless this can keep the more obvious panic by providing an explicit return type for the closure.

Reply:
Fixed